core/main/src/broker/thunder_broker.rs

```diff
@@ -168,6 +249,13 @@ impl ThunderBroker {
         new_response
     }
 
+    fn get_id_from_result(result: &[u8]) -> Option<u64> {
+        if let Ok(data) = serde_json::from_slice::<JsonRpcApiResponse>(result) {
```

Comment:
nitpick: can this be a map() or similar?

Reply:
Changed to

```
        serde_json::from_slice::<JsonRpcApiResponse>(result)
            .ok()
            .and_then(|data| data.id)
```